src/transformers/dynamic_module_utils.py

```diff
@@ -143,9 +144,24 @@ def get_class_in_module(class_name, module_path):
     """
     Import a module on the cache directory for modules and extract a class from it.
     """
-    module_path = module_path.replace(os.path.sep, ".")
-    module = importlib.import_module(module_path)
-    return getattr(module, class_name)
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        module_dir = Path(HF_MODULES_CACHE) / os.path.dirname(module_path)
+        module_file_name = module_path.split(os.path.sep)[-1] + ".py"
+
+        # copy to a temporary directory
+        shutil.copy(f"{module_dir}/{module_file_name}", tmp_dir)
+        cmd = f'import os; os.remove("{module_dir}/{module_file_name}")'
+        os.system(f"python3 -c '{cmd}'")
+        # os.remove(f"{module_dir}/{module_file_name}")
+
+        # copy back the file that we want to import
+        shutil.copyfile(f"{tmp_dir}/{module_file_name}", f"{module_dir}/{module_file_name}")
+
+        # import the module
+        module_path = module_path.replace(os.path.sep, ".")
+        module = importlib.import_module(module_path)
+
+        return getattr(module, class_name)
```

Comment:
Sorry, finally get it:

- we don't need to remove other files (config, `__init__.py`) or `__pycache__` folder
- the point is: we need to remove the `module_file_name` in a subprocess, then copy it back
  - os.system("rm -rf ...") works: as it is in another process
  - os.system(f"python3 -c '{cmd}'"): same, but we don't use Linux specific command --> way to go
  - os.remove(...): not working! I don't know why 😢